Objects/bytearrayobject.c

```diff
@@ -1169,9 +1181,7 @@ bytearray_dealloc(PyObject *op)
                         "deallocated bytearray object has exported buffers");
         PyErr_Print();
     }
-    if (self->ob_bytes != 0) {
-        PyMem_Free(self->ob_bytes);
-    }
+    Py_CLEAR(self->ob_bytes_object);
```

Comment:
```suggestion
    Py_XDECREF(self->ob_bytes_object);
```

no need to set it to NULL here.